backend/src/openarchiefbeheer/zaken/tasks.py

```diff
@@ -58,78 +61,41 @@ def retrieve_and_cache_zaken_from_openzaak() -> None:
             timeout=settings.REQUESTS_DEFAULT_TIMEOUT,
         )
 
-    with selectielijst_api_client or NoClient():
         for index, data in enumerate(data_iterator):
             logger.info("Retrieved page %s.", index + 1)
 
             zaken = data["results"]
-            if selectielijst_api_client:
-                zaken = process_expanded_data(zaken, selectielijst_api_client)
-
-            # Since before we indexed also zaken without einddatum, prevent
-            # crash when retrieving zaken.
-            if zaken_in_db:
+            if not is_full_resync:
+                # Check if we retrieved zaken that are already present in the DB.
+                # If yes, we keep the existing version
                 new_zaken = {zaak["url"]: zaak for zaak in zaken}
                 duplicates = Zaak.objects.filter(url__in=new_zaken.keys())
                 if duplicates.exists():
```

Comment:
Is the exists necessary here if we iterate through the results, without the exists we would also not do anything right since the QuerySet will be empty?